lib/puppet/configurer.rb

```diff
@@ -418,6 +426,40 @@ def save_last_run_summary(report)
     Puppet.log_exception(detail, _("Could not save last run local report: %{detail}") % { detail: detail })
   end
 
+  # Submit updated facts to the Puppet Server
+  #
+  # This method will clear all current fact values, load a fresh set of
+  # fact data, and then submit it to the Puppet Server.
+  #
+  # @return [true] If fact submission succeeds.
+  # @return [false] If an exception is raised during fact generation or
+  #   submission.
+  def resubmit_facts
+    ::Facter.clear
+    facts = find_facts
+
+    saved_fact_terminus = Puppet::Node::Facts.indirection.terminus_class
+    begin
+      Puppet::Node::Facts.indirection.terminus_class = :rest
```

Comment:
`rest` is the Indirector terminus that talks to Puppet Server. We could make this configurable to allow the selection of other terminii, but I'm not sure it would be useful enough to justify the added complexity.